misc/validate_accounting.py

```diff
@@ -309,125 +365,117 @@ def __call__(self, msg):
         self.ok = False
 
 class CheckContext:
-    def __init__(self, accounts, err):
-        # accounts from indexer
-        self.accounts = accounts
+    def __init__(self, err):
         self.err = err
         self.match = 0
         self.neq = 0
         # [(addr, "err text"), ...]
         self.mismatches = []
+        self.lock = threading.Lock()
+
+    def check(self, indexer_account, algod_account):
+        niceaddr = algod_account['address']
+        address = algosdk.encoding.decode_address(niceaddr)
+        xe = ccerr(self.err)
 
-    def check(self, address, niceaddr, microalgos, assets, acfg, appparams, applocal):
-        # check data from sql or algod vs indexer
-        i2v = self.accounts.pop(address, None)
-        if i2v is None:
-            self.neq += 1
-            self.err.write('{} not in indexer\n'.format(niceaddr))
+        microalgos = algod_account['amount-without-pending-rewards']
```

Comment:
missing a few top level fields:
- apps-local-state (not supported by indexer yet)
- rewards (needs some special handling but that is fixed in develop)
- status (need to remove spaces from the algod status)